From eb477d2961aa064bf60e148fa3eae9e7777e040b Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:12:16 -0400
Subject: [powerpc] boot: Add little endian support to elf utils

Message-id: <1410545655-205645-508-git-send-email-dzickus@redhat.com>
Patchwork-id: 94510
O-Subject: [RHEL7 PATCH 507/626] powerpc/boot: Add little endian support to elf utils
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit 002c39dba3fc47b953101790d798f69150366738
Author: Cedric Le Goater <clg@fr.ibm.com>
Date:   Thu Apr 24 09:23:35 2014 +0200

    powerpc/boot: Add little endian support to elf utils

    Signed-off-by: Cedric Le Goater <clg@fr.ibm.com>
    Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/boot/elf_util.c b/arch/powerpc/boot/elf_util.c
index 1567a0c..316552d 100644
--- a/arch/powerpc/boot/elf_util.c
+++ b/arch/powerpc/boot/elf_util.c
@@ -26,7 +26,11 @@ int parse_elf64(void *hdr, struct elf_info *info)
        elf64->e_ident[EI_MAG2]  == ELFMAG2 &&
        elf64->e_ident[EI_MAG3]  == ELFMAG3 &&
        elf64->e_ident[EI_CLASS] == ELFCLASS64 &&
+#ifdef __LITTLE_ENDIAN__
+       elf64->e_ident[EI_DATA]  == ELFDATA2LSB &&
+#else
        elf64->e_ident[EI_DATA]  == ELFDATA2MSB &&
+#endif
        (elf64->e_type            == ET_EXEC ||
         elf64->e_type            == ET_DYN) &&
        elf64->e_machine         == EM_PPC64))
-- 
1.7.1